lib/type_class.ex

```diff
@@ -263,32 +263,44 @@ defmodule TypeClass do
         def __force_type_instance__, do: @force_type_instance
       end
 
+      print_warnings = case System.get_env("PRINT_WARNINGS") do
+        nil -> nil
+        "false" -> false
+        _ -> true
```

Comment:
You have to make sure that your changes agree with each other.

The new default behaviour is to skip compile time tests (FORCE_TYPE_CLASS = true is default).

To match this new behaviour, you have to either:

 - Default PRINT_WARNINGS to false.
 - Rename PRINT_WARNINGS to NOWARN_FORCED_TYPE_CLASS and default it to true (symmetric with the FORCE_TYPE_CLASS).